backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/DatabaseDuplicateCheckService.java

```diff
@@ -0,0 +1,278 @@
+
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DatabaseCourtRepository;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DatabaseDocumentationUnitRepository;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DatabaseDuplicateCheckRepository;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DeviatingDateDTO;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DocumentationUnitDTO;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DocumentationUnitIdDuplicateCheckDTO;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DuplicateRelationDTO;
+import de.bund.digitalservice.ris.caselaw.adapter.transformer.CourtTransformer;
+import de.bund.digitalservice.ris.caselaw.domain.DuplicateCheckService;
+import de.bund.digitalservice.ris.caselaw.domain.DuplicateRelationStatus;
+import de.bund.digitalservice.ris.caselaw.domain.FeatureToggleService;
+import de.bund.digitalservice.ris.caselaw.domain.exception.DocumentationUnitNotExistsException;
+import jakarta.transaction.Transactional;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class DatabaseDuplicateCheckService implements DuplicateCheckService {
+  private final DatabaseDuplicateCheckRepository repository;
+  private final DuplicateRelationService duplicateRelationService;
+  private final DatabaseDocumentationUnitRepository documentationUnitRepository;
+  private final DatabaseCourtRepository databaseCourtRepository;
+
+  private final FeatureToggleService featureToggleService;
+
+  public DatabaseDuplicateCheckService(
+      DatabaseDuplicateCheckRepository duplicateCheckRepository,
+      DuplicateRelationService duplicateRelationService,
+      DatabaseDocumentationUnitRepository documentationUnitRepository,
+      DatabaseCourtRepository databaseCourtRepository,
+      FeatureToggleService featureToggleService) {
+    this.repository = duplicateCheckRepository;
+    this.duplicateRelationService = duplicateRelationService;
+    this.documentationUnitRepository = documentationUnitRepository;
+    this.databaseCourtRepository = databaseCourtRepository;
+    this.featureToggleService = featureToggleService;
+  }
+
+  @Override
+  @Transactional
+  public void checkDuplicates(String docNumber) {
+    try {
+      if (!featureToggleService.isEnabled("neuris.duplicate-check")) {
+        return;
+      }
+
+      var documentationUnit =
+          documentationUnitRepository.findByDocumentNumber(docNumber).orElseThrow();
+
+      var allFileNumbers = collectFileNumbers(documentationUnit);
+      var allEclis = collectEclis(documentationUnit);
+
+      // A duplicate depends on filenumber/ecli
+      if (allFileNumbers.isEmpty() && allEclis.isEmpty()) {
+        return;
```

Comment:
If we have a duplicate via ecli and remove the ecli check we won't delete previously existing duplicates. Instead of returning the whole method, we should return an empty list of found duplicates. (or remove this performance optimization)